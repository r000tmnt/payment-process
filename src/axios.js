import axios from "axios";

const axiosClient = axios.create({
    baseURL: 'https://run.mocky.io/v3/'
})

axiosClient.interceptors.response.use(
    (response) => {
        const res = response?.data || response

        return res;
    },
    async (error) => {
        console.log('api_error :>>>', error.response.data)
        return error.response.data
    }    
)

export default axiosClient