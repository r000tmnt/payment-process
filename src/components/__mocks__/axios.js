import { vi } from "vitest";

export default {
    post: vi.fn()
}