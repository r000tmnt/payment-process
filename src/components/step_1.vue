<template>
    <h1 class="text-center">
        {{ $t(title) }}
    </h1>

    <p class="text-center">{{ $t(desc) }}</p>

    <div class="step-form flex flex-center">
        <div class="step-form-content">
            <label class="step-input-label" for="firstName">
                {{ $t("first-name") }} <span class="require">*</span>
            </label>
            <input class="step-input" type="text" name="firstName" v-model="firstName" required/>
            
            <label class="step-input-label" for="lastName">
                {{ $t("last-name") }} <span class="require">*</span>
            </label>
            <input class="step-input" type="text" name="lastName" v-model="lastName" required/>
            
            <button 
                class="step-form-button flex flex-center text-white" 
                type="button" 
                @click="checkInputValue" 
                :disabled="!firstName.length || !lastName.length" 
                :style="{
                    'backgroundColor': (!firstName.length || !lastName.length)? 'grey' : '#429EF0', 
                    'cursor': (!firstName.length || !lastName.length)? 'unset' : 'pointer',
                    'float': 'right'}">
                <div>{{ $t('continue') }}</div>
                <ChevronRight class="next absolute" fillColor="#ffffff" />
            </button>
        </div>
    </div>

</template>

<script>
import ChevronRight from 'vue-material-design-icons/ChevronRight.vue';

export default{
    props: {
        title: String,
        desc: String,
        fields: Object,
        finish: Boolean,
        result: Object        
    },
    components: {
        ChevronRight
    },
    emits: ['toNext', 'backToPrevious'],
    data(){
        return{
            firstName: '',
            lastName: '',
        }
    },
    methods: {
        async checkInputValue(){
            // console.log('click')
            if(this.firstName.length && this.lastName.length){
                this.returnToParent()
            }
        },

        returnToParent(){
            this.$emit('toNext', { firstName: this.firstName, lastName: this.lastName })
        }        
    },
    created(){
        console.log('fields :>>>', this.$props.fields)

        if(Object.keys(this.$props.fields).length){
            
            // Dynamic assign fields and value
            for(const [key, val] of Object.entries(this.$props.fields)){
                this[`${key}`] = val
            }

        }
    }
}
</script>