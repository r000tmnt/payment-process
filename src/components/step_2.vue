<template>
    <h1 class="text-center">
        {{ $t(title) }}
    </h1>

    <p class="text-center">{{ $t(desc) }}</p>

    <div class="step-form flex flex-center">

        <div v-if="!finish" class="step-input flex-column text-center">

            <div class="step-form-content">
                <CreditCardMultiple :size="48" />
                <h2>{{ $t("credit-card") }}</h2>

                <span style="margin-bottom: 10px;">
                    {{ $t("payment-note") }}
                </span>
                
                <div class="flex flex-around" style="margin: 13px 0 15px 0;">
                    <img class="step-form-img" src="../assets/mastercard_logo.png" alt="mastercard">
                    <img class="step-form-img" src="../assets/visa_logo.png" alt="visa">
                </div>
            </div>

            <div>
                <button 
                    class="step-form-button flex flex-center text-white" 
                    type="button" 
                    @click="$emit('backToPrevious')" 
                    style="background-color: #429EF0; float: left">
                    <ChevronLeft class="prev absolute" fillColor="#ffffff" />
                    <div>{{ $t("back") }}</div>
                </button>    
                
                <button 
                    class="step-form-button flex flex-center text-white" 
                    type="button"  
                    @click="$emit('toNext')"
                    style="background-color: #429EF0; float: right">
                    <div>{{ $t('pay') }}</div>
                    <ChevronRight class="next absolute" fillColor="#ffffff" />
                </button>            
            </div>             
        </div>
        
        <div v-else>
            <div class="step-form-content">
                Congratulations, your payment was successful. 
                Shortly a confirmation email will arrive to your inbox
            </div>            
        </div>
    </div>
</template>

<script>
import ChevronRight from 'vue-material-design-icons/ChevronRight.vue';
import ChevronLeft from 'vue-material-design-icons/ChevronLeft.vue';
import CreditCardMultiple from 'vue-material-design-icons/CreditCardMultiple.vue';

export default{
    props: {
        title: String,
        desc: String,
        finish: Boolean
    },
    components: {
        ChevronRight,
        ChevronLeft,
        CreditCardMultiple
    },
    emits: ['backToPrevious', 'toNext'],
    data(){
        return{

        }
    }
}
</script>