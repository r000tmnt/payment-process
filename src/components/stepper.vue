<template>
    <div id="steps">
        <ul class="flex flex-center">
            <li v-for="(step, index) in steps" :key="index">
                {{ step }}
                {{ Math.round(steps / 2) }}
            </li>
        </ul>
    </div>

    <main></main>
</template>

<script>
export default{
    data(){
        return{
            steps: 2
        }
    }
}
</script>