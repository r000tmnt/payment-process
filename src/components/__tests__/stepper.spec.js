import { describe, it, expect, vi, beforeAll } from 'vitest'

import { mount } from '@vue/test-utils'
import axios from 'axios'
import AppView from '../../App.vue'
import HeaderView from '../header.vue'
import stepper from '../stepper.vue'
import step_1 from '../step_1.vue'
import step_2 from '../step_2.vue'

vi.mock("axios")


const stepperProps = { 
  steps: [
    {
      title: 'data',
      desc: 'data-desc',
      component: step_1,
      finish: false
    },
    {
      title: 'payment',
      desc: 'payment-desc',
      component: step_2,
      finish: false
    }
  ],
  result: {}
} 

describe("StepperView", () => {
  it('Should fill the first name nd last name', async() => {
    const wrapper = mount(AppView, {props: stepperProps})

    const app = wrapper.findComponent(stepper)

    expect(wrapper.html()).toMatchSnapshot()
    expect(app.html()).toMatchSnapshot()

    await app.find('input[name=firstName]').setValue('Rick')
    await app.find('input[name=lastName]').setValue('Lee')
    const nextBtn = wrapper.find('button.step-form-button')

    expect(app.vm.$data.firstName).toBe('Rick')
    expect(app.vm.$data.lastName).toBe('Lee')

    // nextBtn.trigger('click')

    // expect(app.emmited('toNext')).equal
  })
})


// describe('Step 1', () => {
//   it('Return first name and last name', () => {
//     const wrapper = mount(stepper, { 
//       props: stepperProps
//     })

//     const first_name = wrapper.find('input[name=firstName]')
//     const last_name = wrapper.find('input[name=lastName]')
//     const nextBtn = wrapper.find('button.step-form-button')

//     wrapper.setData({
//       firstName: '',
//       lastName: '',      
//     })

//     first_name.setValue('Rick')
//     last_name.setValue('Lee')
//     nextBtn.trigger('click')

//     expect(wrapper.emitted('toNext')).toEqual(
//       { 
//         firstName: 'Rick', 
//         lastName: 'Lee' 
//       }      
//     )

//   })
// })
