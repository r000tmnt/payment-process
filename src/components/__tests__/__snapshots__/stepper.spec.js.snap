// Vitest Snapshot v1

exports[`StepperView > Should fill the first name nd last name 1`] = `
"<header class=\\"flex flex-right\\">
  <div class=\\"button-set\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon help-circle-outline-icon\\" role=\\"img\\"><svg fill=\\"#429EF0\\" class=\\"material-design-icon__svg\\" width=\\"36\\" height=\\"36\\" viewBox=\\"0 0 24 24\\"><path d=\\"M11,18H13V16H11V18M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,6A4,4 0 0,0 8,10H10A2,2 0 0,1 12,8A2,2 0 0,1 14,10C14,12 11,11.75 11,15H13C13,12.75 16,12.5 16,10A4,4 0 0,0 12,6Z\\"><!--v-if--></path></svg></span><button type=\\"button\\" class=\\"header-button\\">help</button></div>
  <div class=\\"button-set\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon account-circle-outline-icon\\" role=\\"img\\"><svg fill=\\"#429EF0\\" class=\\"material-design-icon__svg\\" width=\\"36\\" height=\\"36\\" viewBox=\\"0 0 24 24\\"><path d=\\"M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M7.07,18.28C7.5,17.38 10.12,16.5 12,16.5C13.88,16.5 16.5,17.38 16.93,18.28C15.57,19.36 13.86,20 12,20C10.14,20 8.43,19.36 7.07,18.28M18.36,16.83C16.93,15.09 13.46,14.5 12,14.5C10.54,14.5 7.07,15.09 5.64,16.83C4.62,15.5 4,13.82 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,13.82 19.38,15.5 18.36,16.83M12,6C10.06,6 8.5,7.56 8.5,9.5C8.5,11.44 10.06,13 12,13C13.94,13 15.5,11.44 15.5,9.5C15.5,7.56 13.94,6 12,6M12,11A1.5,1.5 0 0,1 10.5,9.5A1.5,1.5 0 0,1 12,8A1.5,1.5 0 0,1 13.5,9.5A1.5,1.5 0 0,1 12,11Z\\"><!--v-if--></path></svg></span><button type=\\"button\\" class=\\"header-button\\">signin</button></div>
  <div class=\\"button-set\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon flag-variant-outline-icon\\" role=\\"img\\"><svg fill=\\"#429EF0\\" class=\\"material-design-icon__svg\\" width=\\"36\\" height=\\"36\\" viewBox=\\"0 0 24 24\\"><path d=\\"M6,3A1,1 0 0,1 7,4V4.88C8.06,4.44 9.5,4 11,4C14,4 14,6 16,6C19,6 20,4 20,4V12C20,12 19,14 16,14C13,14 13,12 11,12C8,12 7,14 7,14V21H5V4A1,1 0 0,1 6,3M7,7.25V11.5C7,11.5 9,10 11,10C13,10 14,12 16,12C18,12 18,11 18,11V7.5C18,7.5 17,8 16,8C14,8 13,6 11,6C9,6 7,7.25 7,7.25Z\\"><!--v-if--></path></svg></span><button type=\\"button\\" class=\\"header-button\\"></button><span aria-hidden=\\"true\\" class=\\"material-design-icon chevron-down-icon\\" role=\\"img\\"><svg fill=\\"#429EF0\\" class=\\"material-design-icon__svg\\" width=\\"36\\" height=\\"36\\" viewBox=\\"0 0 24 24\\"><path d=\\"M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z\\"><!--v-if--></path></svg></span>
    <!--v-if-->
  </div>
</header>
<div id=\\"steps\\">
  <ul class=\\"flex flex-center\\">
    <li class=\\"flex\\"><span class=\\"step-count\\" style=\\"background-color: rgb(66, 158, 240);\\"><span class=\\"absolute\\">1</span></span><span class=\\"step-title absolute\\">data</span>
      <div class=\\"line\\"></div>
    </li>
    <li class=\\"flex\\"><span class=\\"step-count\\" style=\\"background-color: rgb(66, 158, 240);\\"><span class=\\"absolute\\">2</span></span><span class=\\"step-title absolute\\">payment</span>
      <!--v-if-->
    </li>
  </ul>
</div>
<main class=\\"flex flex-center\\">
  <div id=\\"step-content\\">
    <h1 class=\\"text-center\\">data</h1>
    <p class=\\"text-center\\">data-desc</p>
    <div class=\\"step-form flex flex-center\\">
      <div class=\\"step-form-content\\"><label class=\\"step-input-label\\" for=\\"firstName\\">first-name <span class=\\"require\\">*</span></label><input class=\\"step-input\\" type=\\"text\\" name=\\"firstName\\" required=\\"\\"><label class=\\"step-input-label\\" for=\\"lastName\\">last-name <span class=\\"require\\">*</span></label><input class=\\"step-input\\" type=\\"text\\" name=\\"lastName\\" required=\\"\\"><button class=\\"step-form-button flex flex-center text-white button-disable\\" type=\\"button\\" disabled=\\"\\" style=\\"float: right;\\">
          <div>continue</div><span class=\\"next absolute material-design-icon chevron-right-icon next absolute\\" aria-hidden=\\"true\\" role=\\"img\\"><svg fill=\\"#ffffff\\" class=\\"material-design-icon__svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z\\"><!--v-if--></path></svg></span>
        </button></div>
    </div>
  </div>
</main>"
`;

exports[`StepperView > Should fill the first name nd last name 2`] = `
"<div id=\\"steps\\">
  <ul class=\\"flex flex-center\\">
    <li class=\\"flex\\"><span class=\\"step-count\\" style=\\"background-color: rgb(66, 158, 240);\\"><span class=\\"absolute\\">1</span></span><span class=\\"step-title absolute\\">data</span>
      <div class=\\"line\\"></div>
    </li>
    <li class=\\"flex\\"><span class=\\"step-count\\" style=\\"background-color: rgb(66, 158, 240);\\"><span class=\\"absolute\\">2</span></span><span class=\\"step-title absolute\\">payment</span>
      <!--v-if-->
    </li>
  </ul>
</div>
<main class=\\"flex flex-center\\">
  <div id=\\"step-content\\">
    <h1 class=\\"text-center\\">data</h1>
    <p class=\\"text-center\\">data-desc</p>
    <div class=\\"step-form flex flex-center\\">
      <div class=\\"step-form-content\\"><label class=\\"step-input-label\\" for=\\"firstName\\">first-name <span class=\\"require\\">*</span></label><input class=\\"step-input\\" type=\\"text\\" name=\\"firstName\\" required=\\"\\"><label class=\\"step-input-label\\" for=\\"lastName\\">last-name <span class=\\"require\\">*</span></label><input class=\\"step-input\\" type=\\"text\\" name=\\"lastName\\" required=\\"\\"><button class=\\"step-form-button flex flex-center text-white button-disable\\" type=\\"button\\" disabled=\\"\\" style=\\"float: right;\\">
          <div>continue</div><span class=\\"next absolute material-design-icon chevron-right-icon next absolute\\" aria-hidden=\\"true\\" role=\\"img\\"><svg fill=\\"#ffffff\\" class=\\"material-design-icon__svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z\\"><!--v-if--></path></svg></span>
        </button></div>
    </div>
  </div>
</main>"
`;
